task.go

```diff
@@ -312,7 +313,11 @@ func dispatch(retry bool, taskState *tasks.Task) int {
 		headers = httpRequest.GetHeaders()
 
 		if auth := httpRequest.GetOidcToken(); auth != nil {
-			tokenStr := createOIDCToken(auth.ServiceAccountEmail, httpRequest.GetUrl())
+			audience := httpRequest.GetUrl()
+			if aud := strings.TrimSpace(auth.Audience); aud != "" {
```

Comment:
I think it would be better to move this logic inside the `createOIDCToken` function, as that dealing with those (oidc-specific) details should be its responsibility.
That would also remove the need for the duplication of the emulator test, and the check with/without audience can go in the `oidc_internal_test.go` instead.

Reply:
Do you want me to replace the `serviceAccountEmail` argument inside `createOIDCToken` with `tasks.OidcToken` or just adding a third one called `audience`?

Reply:
A third one called `audience` makes sense to me. That removes the need to propagate this request object (`tasks.OidcToken`) any further.

Reply:
Okay, I'll go for that one. If there should be any more optional arguments one day we could use the options pattern aswell.